### ISSUE TITLE
False positive `used-before-assignment` with multi-item `with` statement and tuple target

### ISSUE THREAD

**Issue by ytoku** [PyCQA/pylint#7545]
### Bug description

The `used-before-assignment` check reports false positive for the combination of multi-item `with` statement and tuple target.

```python
import contextlib


@contextlib.contextmanager
def f():
    yield ("A", "B")


@contextlib.contextmanager
def g(_: str):
    yield "C"


with f() as (a, b), g(a) as c:
    pass
```

### Command used

```shell
pylint --disable=missing-docstring --disable=invalid-name test.py
```

### Pylint output

```shell
************* Module test
test.py:14:22: E0601: Using variable 'a' before assignment (used-before-assignment)

------------------------------------------------------------------
Your code has been rated at 2.86/10 (previous run: 2.86/10, +0.00)
```

### Expected behavior

No error.

### Pylint version

```shell
pylint 2.15.3
astroid 2.12.10
Python 3.10.6 (main, Aug 10 2022, 11:40:04) [GCC 11.3.0]
```

**Comment by ytoku** [comment]
Needs reproduction label?
Do I have to report any additional information?

**Comment by Pierre-Sassoulas** [comment]
No the problem seems legit but we did not check if it reproduce on main (yet).

**Comment by jacobtylerwalls** [comment]
Reproduced at 0d62f67d86c482bd3703e0a75b0f4eb1c8b0e0ee.

@Pierre-Sassoulas probably a good sign that we should leave Needs Triage on until we do meaningful triage and then leave Needs reproduction for more of a "needs info" type deal. Thoughts?

**Comment by Pierre-Sassoulas** [comment]
You're right I'm going to stop doing that.
